git-imerge

```diff
@@ -3640,8 +3653,27 @@ def cmd_reparent(parser, options):
     except ValueError as e:
         sys.exit(e.message)
 
-    sys.stdout.write('%s\n' % (git.reparent(commit_sha1, parent_sha1s),))
+    iterative_reparent(head_sha1, commit_to_reparent_sha1, parent_sha1s, git)
+
+def iterative_reparent(head, commit_to_reparent, parents, git):
+    commit_list = [head];
+    current_commit = head;
+    while current_commit != commit_to_reparent:
```

Comment:
This loop will totally fail (with an `IndexError`, I believe) if `commit_to_reparent` is not a first-parent ancestor of `head`. I understand that this will never be the case the way you want to use the program, but other people might want to use it other ways, and it shouldn't crash for them.

It's not very much harder to rewrite this so that it works for any ancestor of head (not just first-parent ancestors). I'll try to implement that.